tests/robustness/failpoints.go

```diff
@@ -528,3 +529,35 @@ func (f delayPeerNetworkFailpoint) Name() string {
 func (f delayPeerNetworkFailpoint) Available(config e2e.EtcdProcessClusterConfig, clus e2e.EtcdProcess) bool {
 	return config.ClusterSize > 1 && clus.PeerProxy() != nil
 }
+
+type noSpaceFailpoint struct {
+}
+
+func (f noSpaceFailpoint) Inject(ctx context.Context, t *testing.T, lg *zap.Logger, clus *e2e.EtcdProcessCluster) error {
+	member := clus.Procs[rand.Int()%len(clus.Procs)]
+	for member.IsRunning() {
+		lg.Info("Setting up gofailpoint", zap.String("failpoint", f.Name()))
+		err := member.Failpoints().Setup(ctx, f.Name(), "return")
+		if err != nil {
+			lg.Info("goFailpoint setup failed", zap.String("failpoint", f.Name()), zap.Error(err))
+		}
+	}
+
+	return nil
+}
+
+func (f noSpaceFailpoint) Name() string {
+	return "putReturnErrNoSpace"
+}
+
+func (f noSpaceFailpoint) Available(config e2e.EtcdProcessClusterConfig, clus e2e.EtcdProcess) bool {
+	v, err := e2e.GetVersionFromBinary(e2e.BinPath.Etcd)
+	if err != nil {
+		panic(err)
+	}
+	if v.LessThan(version.V3_6) {
```

Comment:
is this sensible? I believe this should be easy to backport to 3.5 and 3.4 however

Reply:
This would not be needed if you used code from `goPanicFailpoint` that checks list of failpoints exposed by gofail

Reply:
added and reused, thanks